explainability-connectors/src/main/java/org/kie/trustyai/connectors/kserve/v2/PayloadParser.java

```diff
@@ -48,7 +52,34 @@ public static PredictionInput rawContentToPredictionInput(ModelInferRequest requ
             default:
                 throw new IllegalArgumentException("Currently unsupported type for Tensor input, type=" + tensor.getDatatype());
         }
+    }
 
+    // this is less efficient than converting everything to a PredictionInput, so avoid when possible
```

Comment:
Are there scenarios where this is still necessary? We can annotate it as `@Deprecated` if we think it should ~never been used

Reply:
Yes, there's one or two times in the codec parser that are forced to use these conversions, hence why they are there